internal/crawler/crawler.go

```diff
@@ -126,7 +125,16 @@ func (c *Crawler) updateProjectInGraph(ctx context.Context, project gitlab.Proje
 	}
 }
 
-func (c *Crawler) handleIncludes(ctx context.Context, project gitlab.Project, filePath string) error {
+func (c *Crawler) handleIncludes(ctx context.Context, project gitlab.Project, filePath string, cycleDetectionMap map[string]bool) error {
```

Comment:
After some thinking and recovering I think this might be a decent solution:

```
diff --git a/internal/crawler/crawler.go b/internal/crawler/crawler.go
index 3913ad4..fbb66f3 100644
--- a/internal/crawler/crawler.go
+++ b/internal/crawler/crawler.go
@@ -130,7 +130,7 @@ func (c *Crawler) updateProjectInGraph(ctx context.Context, project gitlab.Proje
                        return nil
                }
 
-               err := c.handleIncludes(ctx, project, gitlabCIFileName, make(map[string]bool))
+               err := c.handleIncludes(ctx, project, gitlabCIFileName, make(map[string]struct{}))
                if err != nil {
                        c.logger.Error().
                                Err(err).
@@ -141,15 +141,16 @@ func (c *Crawler) updateProjectInGraph(ctx context.Context, project gitlab.Proje
        }
 }
 
-func (c *Crawler) handleIncludes(ctx context.Context, project gitlab.Project, filePath string, cycleDetectionMap map[string]bool) error {
-       if cycleDetectionMap[project.PathWithNamespace] {
+func (c *Crawler) handleIncludes(ctx context.Context, project gitlab.Project, filePath string, cycleDetectionMap map[string]struct{}) error {
+       _, pathTraversed := cycleDetectionMap[project.PathWithNamespace + "--" + filePath]
+       if pathTraversed {
                projectsVisited := make([]string, 0, len(cycleDetectionMap))
                for k := range cycleDetectionMap {
                        projectsVisited = append(projectsVisited, k)
                }
                return errors.New("cycle detected, this should not be possible, the projects visited are: " + strings.Join(projectsVisited[:], ","))
        }
-       cycleDetectionMap[project.PathWithNamespace] = true
+       cycleDetectionMap[project.PathWithNamespace + "--" + filePath ] = struct{}{}
 
        gitlabCIFile, err := c.gitlabClient.GetRawFileFromProject(ctx, project.ID, filePath, project.DefaultBranch)
        if err != nil {
```